hamilton/function_modifiers/macros.py

```diff
@@ -876,3 +856,322 @@ def optional_config(self) -> Dict[str, Any]:
+            :name: post_pipe with globally applied namespace
+
+            @post_pipe(
+                step(_add_one).named("a"), # a
+                step(_add_two, y=source("upstream_node")).named("b"), # foo.b
+                namespace=..., # default -- final_result.a and final_result.b, OR
+                namespace=None, # no namespace -- a and b are exposed as that, OR
+                namespace="foo", # foo.a and foo.b
+            )
+            def final_result(upstream_int: int) -> int:
+                return upstream_int
+
+        In all likelihood, you should not be using this, and this is only here in case you want to expose a node for
+        consumption/output later. Setting the namespace in individual nodes as well as in `post_pipe` is not yet supported.
+    """
+
+    def __init__(
+        self,
+        *transforms: Applicable,
+        namespace: NamespaceType = ...,
+        collapse=False,
+        _chain=False,
+    ):
+        """Instantiates a `@post_pipe` decorator.
+
+        :param transforms: step transformations to be applied, in order
+        :param namespace: namespace to apply to all nodes in the pipe. This can be "..." (the default), which resolves to the name of the decorated function, None (which means no namespace), or a string (which means that all nodes will be namespaced with that string). Note that you can either use this *or* namespaces inside pipe()...
+        :param collapse: Whether to collapse this into a single node. This is not currently supported.
+        :param _chain: Whether to chain the first parameter. This is the only mode that is supported. Furthermore, this is not externally exposed. @flow will make use of this.
+        """
+        super(post_pipe, self).__init__()
+        self.transforms = transforms
+        self.collapse = collapse
+        self.chain = _chain
+        self.namespace = namespace
+
+        if self.collapse:
+            raise NotImplementedError(
+                "Collapsing step() functions as one node is not yet implemented for pipe(). Please reach out if you want this feature."
+            )
+
+        if self.chain:
+            raise NotImplementedError("@flow() is not yet supported -- this is ")
+
+    def transform_node(
+        self, node_: node.Node, config: Dict[str, Any], fn: Callable
+    ) -> Collection[node.Node]:
+        """Injects nodes into the graph.
+
+        We create a copy of the original function and rename it to `function_name_raw` to be the
+        initial node. Then we create a node for each step in `post-pipe` and chain them together.
+        The last node is an identity to the previous one with the original name `function_name` to
+        represent an exit point of `post_pipe`.
+        """
+
+        original_node = node_.copy_with(name=f"{node_.name}_raw")
+
+        def __identity(foo: Any) -> Any:
+            return foo
+
+        transforms = self.transforms + (step(__identity).named(fn.__name__),)
```

Comment:
Edge-case -- if there are no transforms, maybe we shouldn't have identity? I guess the simplest thing would just be to return the node in that case. This would happen if none of them resolve to true...